_posts/2022-04-11-nwaku-update.md

```diff
@@ -0,0 +1,277 @@
+and its interoperability with other Waku v2 clients.
+One major step forward here was adding support for both secure and unsecured WebSocket connections as `libp2p` transports.
+This allows direct connectivity with `js-waku`
+and paves the way for native browser usage.
+We've also added support for parsing and resolving DNS-type `multiaddrs`,
+i.e. multiaddress protocol schemes [`dns`, `dns4`, `dns6` and `dnsaddr`](https://github.com/multiformats/multiaddr/blob/b746a7d014e825221cc3aea6e57a92d78419990f/protocols.csv#L8-L11).
+A `nwaku` node can now also be [configured with its own IPv4 DNS domain name](https://github.com/status-im/nim-waku/tree/d2fccb5220144893f994a67f2cc26661247f101f/waku/v2#configuring-a-domain-name) 
+allowing dynamic IP address allocation without impacting a node's reachability by its peers.
+
+## 4. Peer discovery
+
+_Peer discovery_ is the method by which nodes become aware of each other’s existence.
+The question of peer discovery in a Waku v2 network has been a focus area since the protocol was first conceptualized.
+Since then several different approaches to discovery have been proposed and investigated.
+We've implemented three discovery mechanisms in `nwaku` so far:
+
+### DNS-based discovery
+
+`nwaku` nodes can retrieve an authenticated, updateable list of peers via DNS to bootstrap connection to a Waku v2 network.
+Our implementation is based on [EIP-1459](https://eips.ethereum.org/EIPS/eip-1459).
+
+### GossipSub peer exchange
+
+[GossipSub Peer Exchange (PX)](https://github.com/libp2p/specs/blob/10712c55ab309086a52eec7d25f294df4fa96528/pubsub/gossipsub/gossipsub-v1.1.md#prune-backoff-and-peer-exchange) is a GossipSub v1.1 mechanism
+whereby a pruning peer may provide a pruned peer with a set of alternative peers
+where it can connect to reform its mesh.
+This is a very suitable mechanism to gradually discover more peers
+from an initial connection to a small set of bootstrap peers.
+It is enabled in a `nwaku` node by default.
+
+### Waku Node Discovery Protocol v5
+
+This is a DHT-based discovery mechanism adapted to store and relay _node records_.
+Our implementation is based on [Ethereum's Discovery v5 protocol](https://github.com/ethereum/devp2p/blob/fa6428ada7385c13551873b2ae6ad2457c228eb8/discv5/discv5-theory.md)
+with some [minor modifications](https://rfc.vac.dev/spec/33/) to isolate our discovery network from that of Ethereum.
+
+## 5. Spam protection using RLN
+
+An early addition to our suite of protocols was [an extension of `11/WAKU-RELAY`](https://rfc.vac.dev/spec/32/)
+that provided spam protection using [Rate Limiting Nullifiers (RLN)](https://rfc.vac.dev/spec/32/).
+The `nwaku` client now contains a working demonstration and integration of RLN relay.
+Check out [this tutorial](https://github.com/status-im/nim-waku/blob/ee96705c7fbe4063b780ac43b7edee2f6c4e351b/docs/tutorial/rln-chat2-live-testnet.md) to see the protocol in action using a toy chat application built on `nwaku`.
```

Comment:
Mention that we'd love to see people join dogfooding of this as part of operator incentive testnet, call to action to join discord here